modules/consensus/accept_test.go

```diff
@@ -286,6 +311,181 @@ func TestUnitValidateHeader(t *testing.T) {
 	}
 }
 
+// TestCheckHeaderTarget probes the checkHeaderTarget function and checks that
+// the result matches the result of checkTarget.
+func TestCheckHeaderTarget(t *testing.T) {
+	var b types.Block
+	var h types.BlockHeader
+
+	tests := []struct {
+		target   types.Target
+		expected bool
+		msg      string
+	}{
+		{types.RootDepth, true, "checkHeaderTarget failed for a low target"},
+		{types.Target{}, false, "checkHeaderTarget passed for a high target"},
+		{types.Target(h.ID()), true, "checkHeaderTarget failed for a same target"},
```

Comment:
huh. I've always done this by manually mining a target that either did or didn't fit the criteria. The testing target is {128}, which means you have an exactly 50% chance of finding a valid header each attempt. Not too painful to write a loop that guarantees the correct outcome, it's cryptographically guaranteed to complete within 128 iterations.

The way you've done it is fine as well